code/__HELPERS/heap.dm

```diff
@@ -11,6 +11,12 @@
 	L = new()
 	cmp = compare
 
+/datum/heap/Destroy(force, ...)
+	for(var/i in L) // because this is before the list helpers are loaded
+		qdel(i)
+	L = null
+	return ..()
```

Comment:
Whats the reason for qdeleting everything in the list?

I worry this will have really bad side effects if we use heaps to do other logic with atom sorting in the future.

Reply:
Nixing that from the port. TG did kinda pingeonhole themselves in how heaps can be used.

Still your responsibility to clear out the list in the future.